### ISSUE TITLE
Calendar doesn't refresh events on real time [Milestone 6]

### ISSUE THREAD

**Issue by jorge9200** [Inevio/GoogleCalendar#4]
We provide via API with some events everytime changes happens on that gcalendar account. You have to refresh those changes on horbito app on real time. I've tried to create a new event on gcalendar native app, and nothing can be seen on horbito app until I restart it.

It has to react on real time to these changes

**Comment by umair-brainx** [comment]
The real time syncing data was not part of initial requirements.

**Comment by jorge9200** [comment]
Yes it was. We asked for entire calendar functionality and obviously real time syncing is an important part of it. We develop a event system from start to provide all the necessary things to accomplish this.

Anyway it is really simple from your side, you only need to add this to your code:

`api.integration.gcalendar.on( ‘modified’, function( data ){
    //handle
})`
`api.integration.gcalendar.on( ‘removed’, function( data ){
    //handle
})`

So please take a look, if you have any problem we can discuss it with Adeel but we need this working.

**Comment by umair-brainx** [comment]
@jorge9200 Sorry. I totally disagree with you. It was never part of requirements and have not mentioned in any of your design and requirements before.

**Comment by jorge9200** [comment]
As you can look on freelancer project, we especified to implement notification management and with notifications we refer to all posible notifications the api give us, such as modifies and removes.

![tg_image_737365860](https://user-images.githubusercontent.com/8724844/34771310-6401d8b0-f604-11e7-85d2-aabfd33f6c2f.jpeg)

So it was part of requirements from the start of project.

**Comment by umair-brainx** [comment]
This is what you have proposed not what we have agreed.
Here is what we have agreed.

Milestone 1:
-Login Screen [View and Functionality]
-Calendar Day View Screen [View]
-Side bar and Search bar [View]

Milestone 2: 
-Calendar Day View Screen [Functionality]
-Side bar and Search bar [Functionality]

Milestone 3:
-Calendar Week View Screen [View and Functionality]
-Calendar Month View Screen [View and Functionality]

Milestone 4:
-Create New Event Screen [View and Functionality]
-Create New Calendar Screen [View and Functionality]

Milestone 5
-Event popup  Details  Screen [View and Functionality]
-Edit Event Screen [View and Functionality]
-Testing, Polishing and making it ready to go live.


Thats nowhere include the badges, popups and notifications.

**Comment by Robby2023** [comment]
Wow, it really is a constant fight with you guys to get things done... It was clear in the project description and you "deleted" it deliberately in your milestones so you didn't have to develop it? That's a pretty huge deal... @umair-brainx 

I'm going to contact Adeel to see what's going on.

**Comment by umair-brainx** [comment]
Sure @Robby2023 . You are welcome to contact Adeel. To clear your doubt I don't run off from work instead these two screens
-Event popup Details Screen [View and Functionality]
-Edit Event Screen [View and Functionality]

have been added to project on my suggestions. You guys were not even aware that you are missing important part of actual flow.

**Comment by umair-brainx** [comment]
I have implemented the remove and edit event syncing. But add event is remaining as i don't receive the ID of calendar to whom i should add event

**Comment by jorge9200** [comment]
there is no way to get a calendar from an event information, so its a google api limitation.

We will have to save in program var the events are showing with their respective calendar. And when some event reach to us, check on that list and update it if you found it.

**Comment by umair-brainx** [comment]
I am doing the suggested way for **edit** and **remove** but for **new** for new event its sure that it will not be already in our program events list so how can we get the calendarId for that?

**Comment by jorge9200** [comment]
you are totally right, I need to talk with @icastillejogomez then

**Comment by jorge9200** [comment]
Ok. We have investigate everything and this is our final proposition:

We are going to add a refresh button to the app. If you are not disposed to do it, we will add an extra milestone to the app.

![image](https://user-images.githubusercontent.com/8724844/37208205-bf89f6a8-23a0-11e8-871f-ca774917d486.png)
![refresh-icon 2x](https://user-images.githubusercontent.com/8724844/37208219-cd442e9e-23a0-11e8-9b4b-65cbb684f72c.png)
![refresh-icon](https://user-images.githubusercontent.com/8724844/37208220-cd5cb194-23a0-11e8-8e6a-b7875f4c5d5c.png)

The button only will refresh the events showed on the current view (current month, current week or current day), this will be done calling a **refeshEventsOnView**() function (the name is just an example)

Also every minute (define this as a variable to be change if necessary later) you will call the **refeshEventsOnView**() function

Also,

- If you receive an event and its on showed view, you redraw it (no request)
- If you receive a 'new event' event you do nothing

**Comment by jorge9200** [comment]
Moved to Milestone 7 Remaining:

- [ ] Sync events function
- [ ] Sync-button UI
- [ ] Call sync every minute (let the minute in a var to change it later)
- [ ] Call sync when you change interval
- [ ] Call sync on sync-button

**Comment by umair-brainx** [comment]
Fixed.
